app/controllers/admin/categories_controller.rb

```diff
@@ -14,17 +15,22 @@ def new
 
   def update
     if @category.respond_to?(category_params[:field])
-      @valid = @category.update_attributes(category_params[:field]=> category_params[:value])
+      @valid = @category.update_attributes(category_params[:field] => category_params[:value])
       flash.now[:error] = @category.errors.full_messages.join(',') unless @valid
     else
-      flash.now[:error] = I18n.t('models.categories.no_column')
+      flash.now[:error] = I18n.t('models.categories.unknown_column')
     end
   end
 
   def destroy
-    redirect_to categories_path, @category.try(:destroy) ? { notice: I18n.t('controllers.categories.destroyed', :category => @category.id) } : { error: I18n.t('controllers.categories.not_destroyed', :category => @category.id) }
+    redirect_to admin_categories_path, @category.destroy ? { notice: I18n.t('controllers.categories.destroyed', :category => @category.id) } : { error: I18n.t('controllers.categories.not_destroyed', :category => @category.id) }
   end
 
+  protected
+    def restrict_user_by_role
```

Comment:
looks like this before filter is the same as you have for checking admins in application controller